File: C++/First Steps in Coding - Basics of C++/Conditional Statements/Exercises/Shopping.cpp
#include <iostream>
#include <iomanip>
#include <cmath>
#include <cstdlib>
using namespace std;
int main() {
	double videoCardPrice = 250, cpuPrice, ramPrice, requiredSum, budget;
	int videoCards, cpus, rams;
	cin >> budget; cin >> videoCards; cin >> cpus; cin >> rams;
	cpuPrice = (videoCardPrice * videoCards) * 0.35; ramPrice = (videoCardPrice * videoCards) * 0.1;
	requiredSum = (videoCards * videoCardPrice) + (cpus * cpuPrice) + (rams * ramPrice);
	if (videoCards > cpus) {
		requiredSum -= (requiredSum * 0.15);
	}
	if (requiredSum <= budget) {
		cout << fixed << setprecision(2) << "You have " << budget - requiredSum << " leva left!";
	}
	else {
		cout << fixed << setprecision(2) << "Not enough money! You need " << requiredSum - budget << " leva more!";
	}
}

File: Java/Conditional Statements/Exercise/07. Shopping/Shopping.java
import java.util.Scanner;

public class Shopping {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        double budget = Double.parseDouble(input.nextLine());
        int videoCards = Integer.parseInt(input.nextLine());
        int processors = Integer.parseInt(input.nextLine());
        int rams = Integer.parseInt(input.nextLine());

        //prices
        double gpuPrice = 250, cpuPrice = ((videoCards * gpuPrice) * 0.35), ramPrice = ((videoCards * gpuPrice) * 0.1);

        double totalExpenses = (videoCards * gpuPrice) + (processors * cpuPrice) + (rams * ramPrice);

        if(videoCards > processors){
            totalExpenses -= totalExpenses * 0.15;
        }

        if(totalExpenses > budget) {
            System.out.printf("Not enough money! You need %.2f leva more!", totalExpenses - budget);
        }else {
            System.out.printf("You have %.2f leva left!", budget - totalExpenses);
        }
    }
}

/*
⦁	Видеокарта – 250 лв./бр.
        ⦁	Процесор – 35% от цената на закупените видеокарти/бр.
        ⦁	Рам памет – 10% от цената на закупените видеокарти/бр.
*/

/*
⦁	Бюджетът на Петър - реално число в интервала [0.0…100000.0]
        ⦁	Броят видеокарти - цяло число в интервала [0…100]
        ⦁	Броят процесори - цяло число в интервала [0…100]
        ⦁	Броят рам памет - цяло число в интервала [0…100]
*/

//example input
/*
900
2
1
3
*/
